### PR TITLE
Add new administrator sparql interface

### PR DESCRIPTION
**Pull request by cjmyers**
New SPARQL update panel on admin page.

### CODE REVIEW

File: lib/app.js [hunk 1]
```diff
@@ -283,6 +285,22 @@ function App () {
   app.post('/admin/federate', requireAdmin, bodyParser.urlencoded({ extended: true }), actions.admin.federate)
   app.post('/admin/retrieveFromWebOfRegistries', requireAdmin, actions.admin.retrieve)
 
+  app.get('/admin/sparql', requireAdmin, sparqlAdmin)
+  app.post('/admin/sparql', requireAdmin, bodyParser.urlencoded({ extended: true }), sparqlAdmin)
+
+  function sparqlAdmin (req, res) {
```
Comment (zachzundel) [inline lib/app.js]: Can this get moved down near the sparql function it is copied from?

Comment (cjmyers) [reply lib/app.js]: Sure.  I actually moved it up to be with the other admin pages though.  Which do you think is better?

File: lib/views/admin/sparql.js [hunk 2]
```diff
@@ -47,3 +75,57 @@ module.exports = function (req, res) {
...
+  } catch (e) {
+    form(req, res, {
+      query: req.body.query,
+      graph: req.body.graph,
+      errors: [
+        e.stack
+      ]
+    })
+
+    return
```
Comment (zachzundel) [inline lib/views/admin/sparql.js]: This isn't necessary

File: lib/views/admin/sparql.js [hunk 3]
```diff
@@ -47,3 +75,57 @@ module.exports = function (req, res) {
     })
   }
 }
+
+function post (req, res) {
+  req.setTimeout(0) // no timeout
+
+  var graphUri
```
Comment (zachzundel) [inline lib/views/admin/sparql.js]: If we declare and them immediately set (like we do here) this can be combined.

File: lib/app.js (hunk 1 above)
Comment (zachzundel) [reply lib/app.js]: Wherever they are (I am partial to below all the routes, but this is a preference) they should be together, I think.

Comment (cjmyers) [reply lib/app.js]: Ah, you mean just the function?  Yes, I can move that.

Comment (zachzundel) [reply lib/app.js]: Yep, just the `sparqlAdmin` function.

**Comment by cjmyers** [comment]
Made the suggested changes.

**Pull request merged**